src/transactions/feed/NftFeedItem.tsx

```diff
@@ -12,12 +16,45 @@ import { Nft } from 'src/nfts/types'
 import useSelector from 'src/redux/useSelector'
 import { getFeatureGate } from 'src/statsig'
 import { StatsigFeatureGates } from 'src/statsig/types'
+import colors from 'src/styles/colors'
 import fontStyles from 'src/styles/fonts'
 import variables from 'src/styles/variables'
 import { NftTransfer, TokenTransactionTypeV2 } from 'src/transactions/types'
 import networkConfig from 'src/web3/networkConfig'
 import { walletAddressSelector } from 'src/web3/selectors'
 
+function NftIconPlaceholder({ testID = 'NftIconPlaceHolder' }: { testID?: string }) {
+  return (
+    <SkeletonPlaceholder>
+      <Svg testID={testID} width={40} height={40}>
+        <Circle cx={20} cy={20} r={20} fill={colors.gray3} />
+      </Svg>
+    </SkeletonPlaceholder>
+  )
+}
+
+function NftIcon({ nft }: { nft: Nft }) {
+  const [loading, setLoading] = useState(true)
+  const [imageLoadingError, setImageLoadingError] = useState(false)
+  return imageLoadingError ? (
+    <View style={[styles.circleIcon, styles.errorCircleIcon]}>
+      <ImageErrorIcon size={30} testID="NftFeedItem/NftErrorIcon" />
+    </View>
+  ) : (
+    <FastImage
```

Comment:
i think there might be some way we can abstract this fastImage component, we have it in quite a few places now and they all do the same thing with slightly different styles. we can address this in a separate PR, but i think it'd be a nice improvement to handle the loading / error logic in a single place